disaggregated_memory/configs/riscv-main.py

```diff
@@ -0,0 +1,293 @@
+)
+
+# Here we setup the processor. We use a simple processor.
+processor = SimpleProcessor(cpu_type=cpu_type, isa=ISA.RISCV, num_cores=1)
+
+# Here we setup the board which allows us to do Full-System ARM simulations.
+board = RiscvComposableMemoryBoard(
+    clk_freq=args.cpu_clock_rate,
+    processor=processor,
+    local_memory=local_memory,
+    remote_memory=remote_memory,
+    cache_hierarchy=cache_hierarchy,
+)
+
+# commands to execute to run the simulation.
+mount_cmd = ["mount -t sysfs - /sys;", "mount -t proc - /proc;"]
+
+warn("The command list to execute has to be manually set!")
+
+local_stream = [
+    'echo "starting STREAM locally!";',
+    "numastat;",
+    "numactl --membind=0 -- "
+    + "/home/ubuntu/simple-vectorizable-benchmarks/stream/"
+    + "stream.hw.m5 10000000;",
+    "numastat;",
+]
+
+interleave_stream = [
+    'echo "starting interleaved STREAM!";',
+    "numastat;",
+    "numactl --interleave=0,1 -- "
+    + "/home/ubuntu/simple-vectorizable-benchmarks/stream/"
+    + "stream.hw.m5 10000000;",
+    "numastat;",
+]
+
+remote_stream = [
+    'echo "starting STREAM remotely!";',
+    "numastat;",
+    "numactl --membind=1 -- "
+    + "/home/ubuntu/simple-vectorizable-benchmarks/stream/"
+    + "stream.hw.m5 10000000;",
+    "numastat;",
+]
+
+# Since we are using kvm to boot the system, we can boot the system with
+# systemd enabled!
+cmd = (
+    mount_cmd
+    + ["m5 --addr=0x10010000 exit;"]
+    + local_stream
+    + interleave_stream
+    + remote_stream
+    + ["m5 --addr=0x10010000 exit;"]
+)
+
+workload = CustomWorkload(
+    function="set_kernel_disk_workload",
+    parameters={
+        "disk_image": DiskImageResource(
+            local_path="/home/kaustavg/disk-images/rv64gc-hpc-2204.img",
+            root_partition="1",
+        ),
+        "kernel": CustomResource(
+            "/home/kaustavg/kernel/gem5-resources/src/riscv-fs/riscv64-sample/bbl"
+        ),
+        "readfile_contents": " ".join(cmd),
+    },
+)
+
+ckpt_to_read_write = ""
+if args.ckpt_file != "":
+    ckpt_to_read_write = (
+        m5.options.outdir + "/" + args.ckpt_file + str(args.instance)
+    )
+    # inform the user where the checkpoint will be saved
+    print("Checkpoint will be saved in " + ckpt_to_read_write)
+else:
+    warn("A checkpoint path was not provided!")
+
+# This disk image needs to have NUMA tools installed.
+board.set_workload(workload)
+
+# This script will boot two NUMA nodes in a full system simulation where the
+# gem5 node will be sending instructions to the SST node. the simulation will
+# after displaying numastat information on the terminal, which can be viewed
+# from board.terminal.
+board._pre_instantiate()
+root = Root(full_system=True, board=board)
+board._post_instantiate()
+
+
+# define on_exit_event
+def handle_exit():
+    yield True  # Stop the simulation. We're done.
+
+
+# Here are the different scenarios:
+# no checkpoint, run everything in gem5
+if args.take_ckpt == "True":
```

Comment:
We can find a better way to communicate "is this executing in SST".

Reply:
But the user might also run gem5-only experiments, right? Or should we completely remove that option from these runscipts?